lib/AST/ArchetypeBuilder.cpp

```diff
@@ -2087,7 +2090,10 @@ Type ArchetypeBuilder::mapTypeOutOfContext(ModuleDecl *M,
   if (!canType->hasArchetype())
     return type;
 
-  assert(genericParams && "dependent type in non-generic context");
+  if (!genericParams) {
+    // Failed: dependent type in non-generic context.
+    return Type();
```

Comment:
Yeah, so returning a null type isn't really any better than asserting -- you're just pushing the crash downstream, where it is harder to figure out the root case.

Which crasher regressed and what was the error?